src/fights.ts

```diff
@@ -1428,6 +1428,62 @@ const freeFightSources = [
     },
     true
   ),
+  // Use NC forcers to complete shadow quests
+  new FreeFight(
```

Comment:
Is this actually a `FreeFight`? I personally don't think this feature belongs in fights.ts - it should be handled either by potion.ts or mood.ts since it deals with obtaining buffs

Reply:
Alternatively this should be handled directly in the free shadow rift fights - i.e. if the conditions below return `true`, then we should spike a free rift monster -> grab the buff -> spike a free rift monster -> grab the buff etc.

Notably, this requires that you have not obtained Shadow Affinity for the day, so it makes sense to have alternative handling of using spikes to obtain the buff (but you'd have to think through how we are going to make those spike fights free)

Reply:
It's kind of weird, since we automatically obtain shadow affinity when we take the quest, we either have to do parka spikes with the buff active or after it's run out.

I'm thinking of just using spikes with shadow affinity turns, but there's probably a better solution.